### ISSUE TITLE
Low shared memory

### ISSUE THREAD

**Issue by depshad** [pytorch/pytorch#45728]
I’m unable to use num_workers>0 in Pytorch Dataloader. The reason is the low shared memory available (64M). 
Can we re-direct or override the path used internally **(/dev/shm)** by Pytorch?

Example : %env JOBLIB_TEMP_FOLDER=/tmp for Sklearn
Reference for sklearn :  https://stackoverflow.com/questions/40115043/no-space-left-on-device-error-while-fitting-sklearn-model

Please suggest.

**Comment by SsnL** [comment]
Shared memory is not the same as a temp folder. It is not something that you can just substitute with a new path.

**Comment by colesbury** [comment]
No, there's no equivalent. Sklearn uses /dev/shm simply for fast temporary storage. The PyTorch data loader needs shared memory for inter-process communication.

You can try switching to the "file system" method [`torch.multiprocessing.set_sharing_strategy('file_system')`](https://pytorch.org/docs/stable/multiprocessing.html#torch.multiprocessing.set_sharing_strategy). My guess is that it won't matter, but it might be worth trying.

You can also try increasing the size of /dev/shm: https://masukkhan.wordpress.com/2015/12/09/resize-devshm-filesystem-in-linux/

**Comment by depshad** [comment]
Unfortunately, in my case increasing shared memory space is not possible. I will look into the 'file system' method suggested.

**Comment by XinDongol** [comment]
any update on this thread? 

Increasing shared memory space is not possible neither in my case

**Comment by sevmag** [comment]
I'm running into the same problem does anyone have a fix for this?
